### PR TITLE
add file list to index detail view

### PR DESCRIPTION
**Pull request by ReeceHoffmann**
New index detail view shown below.

image

Updated image 10/01/22: Changed wording to requested

### CODE REVIEW

**Review by igboyes — CHANGES_REQUESTED** [review]
> Data files available to workflows using this index.

File: src/js/indexes/components/Files.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { map } from "lodash-es";
+import React, { useMemo } from "react";
+import { connect } from "react-redux";
+import { Badge, BoxGroup, BoxGroupHeader } from "../../base";
+import { File } from "./File";
+
+export function Files({ files }) {
+    const fileComponents = useMemo(
+        () =>
+            map(files, file => (
```
Comment (igboyes) [inline src/js/indexes/components/Files.js]: Prefer object spread here.
```js
map(files, ({ name, type }) => ())
```

File: src/js/indexes/components/General.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import React from "react";
 import Contributors from "./Contributors";
 import IndexOTUs from "./OTUs";
+import Files from "./Files";
 
 export const IndexGeneral = () => (
     <div>
```
Comment (igboyes) [inline src/js/indexes/components/General.js]: Can we use a fragment here instead of `div`?

File: src/js/indexes/components/File.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const StyledFile = styled(BoxGroupSection)`
+    display: flex;
+`;
+
+const StyledSize = styled.div`
+    margin-left: auto;
+    font-weight: ${getFontWeight("bold")};
+`;
+
+export function File({ fileName, download_url, id, size }) {
```
Comment (igboyes) [inline src/js/indexes/components/File.js]: Looks like there is no universal preference for declarations vs arrow functions. However, let's be consistent within Virtool. Almost everything is an arrow function right now, so let's stick with that.

File: src/js/indexes/components/Files.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { map } from "lodash-es";
+import React, { useMemo } from "react";
+import { connect } from "react-redux";
+import { Badge, BoxGroup, BoxGroupHeader } from "../../base";
+import { File } from "./File";
+
+export function Files({ files }) {
```
Comment (igboyes) [inline src/js/indexes/components/Files.js]: Needs a test.

File: src/js/indexes/components/Files.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    const fileComponents = useMemo(
```
Comment (igboyes) [inline src/js/indexes/components/Files.js]: Use of `useMemo` is not advisable here.

1. Unnecessary re-render will already be prevented by `react-redux`. No render occurs unless the values returned from `mapStateToProps` change.
2. Conventional React wisdom: optimize when required, not by deefault. There will be little benefit to `useMemo` here even without `react-redux`. The added complexity and memory usage is not worth it.

I didn't know `useMemo` could return components. Good to know!

**Review by igboyes — CHANGES_REQUESTED** [review]

File: src/js/indexes/components/Files.js (hunk 5 above)
Comment (ReeceHoffmann) [reply src/js/indexes/components/Files.js]: Good points, changed.

File: src/js/indexes/components/File.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    font-weight: ${getFontWeight("bold")};
+`;
+
+export const File = ({ fileName, download_url, size }) => {
+    return (
+        <StyledFile>
+            <a href={download_url}>{fileName}</a> <StyledSize>{byteSize(size)}</StyledSize>
+        </StyledFile>
+    );
+};
```
Comment (igboyes) [inline src/js/indexes/components/File.js]:
```suggestion
export const File = ({ fileName, download_url, size }) => (
    <StyledFile>
        <a href={download_url}>{fileName}</a> <StyledSize>{byteSize(size)}</StyledSize>
    </StyledFile>
);
```

File: src/js/indexes/components/__tests__/Files.test.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        renderWithProviders(<Files {...props} />);
+        expect(screen.getByText("Files")).toBeInTheDocument();
+    });
+
+    it("Render Names", () => {
+        renderWithProviders(<Files {...props} />);
+        expect(screen.getByText("foo")).toBeInTheDocument();
+    });
+
+    it("Render sizes", () => {
```
Comment (igboyes) [inline src/js/indexes/components/__tests__/Files.test.js]:
```suggestion
    it("should render Names", () => {
        renderWithProviders(<Files {...props} />);
        expect(screen.getByText("foo")).toBeInTheDocument();
    });

    it("should render sizes", () => {

```

File: src/js/indexes/components/__tests__/Files.test.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import { Files } from "../Files";
+import { screen } from "@testing-library/react";
+
+describe("<IndexDetail />", () => {
+    let props;
+
+    beforeEach(() => {
+        props = {
+            files: [
+                { id: "1", name: "foo", DownloadUrl: "https://Virtool.ca/testUrl/foo", size: "1024" },
```
Comment (igboyes) [inline src/js/indexes/components/__tests__/Files.test.js]: Is `DownloadUrl` correct?

File: src/js/indexes/components/File.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+const StyledFile = styled(BoxGroupSection)`
+    display: flex;
+`;
+
+const StyledSize = styled.div`
+    margin-left: auto;
+    font-weight: ${getFontWeight("bold")};
+`;
+
+export const File = ({ fileName, download_url, size }) => {
```
Comment (igboyes) [inline src/js/indexes/components/File.js]: I think there is opportunity to share this with the file list in the subtraction detail view. Can you look at doing this?

**Review by igboyes — CHANGES_REQUESTED** [review]

**Review by igboyes — APPROVED** [review]

**Pull request merged**